### PR TITLE
Translations update from Hosted Weblate

### PR DESCRIPTION
**Pull request by weblate**
Translations update from [Hosted Weblate](https://hosted.weblate.org) for [LinuxCNC/LinuxCNC](https://hosted.weblate.org/projects/linuxcnc/linuxcnc/).

It also includes following components:

* [LinuxCNC/LinuxCNC/Gmocappy](https://hosted.weblate.org/projects/linuxcnc/gmocappy/)

Current translation status:

Weblate translation status

### CODE REVIEW

**Comment by rene-dev** [comment]
I don't agree with XY-Drehung and Arbeitsoffset. Lathe Wear Offsets is more like Werkzeugverschleiß.
Rotation of Z I would translate as "Rotation um Z"

File: src/po/gmoccapy/de.po [hunk 1]
```diff
@@ -1917,8 +1917,9 @@ msgid "Rot"
 msgstr "Rot"
 
 #: ../../../../lib/python/gladevcp/offsetpage.glade.h:14
+#, fuzzy
 msgid "Rotation of Z"
-msgstr "Drehung um Z"
```
Comment (hansu) [inline src/po/gmoccapy/de.po:1921]: > @smoe I don't agree with XY-Drehung and Arbeitsoffset. Lathe Wear Offsets is more like Werkzeugverschleiß. Rotation of Z I would translate as "Rotation um Z"

It seems a bit like a bug in Weblate. I changed this string at the same session where I also made the other changes.
(Look at the time and author in the header) But this change was overwritten again by the old one. May Steffen was logged in in Weblate at the same time?

File: src/po/gmoccapy/de.po [hunk 2]
```diff
@@ -132,8 +132,8 @@ msgid ""
 "    set\n"
 "selected"
 msgstr ""
-"    gewähltes\n"
-"   setzen"
+"    Gewählten\n"
+"Offset setzen"
```
Comment (hansu) [inline src/po/gmoccapy/de.po:137]: And also I noticed that it is not possible to remove those preceding spaces in the translated string if the source string have these :unamused:

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: These warnings about blanks are indeed unfortunate. The purpose like was to center to words but when the lengths of the translations don't fit then this is very much undoable. I also think that this is something we should look into in the source tree....later :o)

Comment (hansu) [reply src/po/gmoccapy/de.po:137]: I don't care about the warnings. But if i remove the spaces, press save, then Weblate adds the spaces again. That is annoying. I wonder if that can be adjusted somewhere in the project settings.

File: src/po/gmoccapy/de.po (hunk 1 above)
Comment (smoe) [reply src/po/gmoccapy/de.po:1921]: Just correct any nonsense of mine, please. Literally the "Rotation of Z" is more like a "Rotation von Z", i.e. a change of direction of Z while keeping the unit length. I admit that I do not know how this makes any sense if X and Y do not turn with it. Weblate finds it on https://hosted.weblate.org/translate/linuxcnc/linuxcnc/de/?q=Drehung&sort_by=-priority%2Cposition&offset=14 . "Drehung um Z" würde "Rotation around Z" im Englischen sein?!?

Comment (smoe) [reply src/po/gmoccapy/de.po:1921]: I admit not to have any idea what an XY-Rotation from https://hosted.weblate.org/translate/linuxcnc/linuxcnc/de/?offset=1&q=XY-Drehung&sort_by=-priority%2Cposition&checksum= could possibly be if not an XY-Drehung :) Again, please override.

Comment (smoe) [reply src/po/gmoccapy/de.po:1921]: I removed the Arbeitsoffsets. Abnutzungs-Kompensation? Anyway - someone will know the right word.

File: src/po/gmoccapy/de.po (hunk 2 above)
Comment (smoe) [reply src/po/gmoccapy/de.po:137]: I often think that the original leading spaces are erroneous. But as long as these are present, I suggest the translation should also live with them.

Comment (hansu) [reply src/po/gmoccapy/de.po:137]: Of course the original leading spaces are not erroneous. But as we have a different word order in other languages it makes sense to have that alignment-spaces at another position. Consider this example:
```
msgid ""
"    Zero\n"
"Rotational"
msgstr ""
"    Drehachsen\n"
"nullen"
```

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: You mean
```
msgstr ""
"Drehachsen\n"
"    nullen"
'''
```
which then triggers the weblate warnings. And what we would want is to just write
```
msgstr ""
"Drehachsen nullen"
'''
```
and somewhere in the embedding code there is something equivalent to `align="center"`. I have not looked at the respective source code and decided for myself to just live with how it is now. The Google Summer of Code just requested projects to apply. Maybe this is something?

Comment (hansu) [reply src/po/gmoccapy/de.po:137]: > You mean
> 
> ```
> msgstr ""
> "Drehachsen\n"
> "    nullen"
> '''
Yes that is the goal, but Weblate always adds those leading spaces.
   




> ```
> msgstr ""
> "Drehachsen nullen"
> '''
> ```
> 
> and somewhere in the embedding code there is something equivalent to `align="center"`. 

For this you have to create an extra label for that in the code and enable line break if it doesn't fit the width.

A cleaner solution would be to change the source string to
```
msgid ""
"Zero\n"
"Rotational"
```
and apply a CSS setting on it to right align the label. But still needless work if Weblate wouldn't be so stupid ;-)

Comment (hansu) [reply src/po/gmoccapy/de.po:137]: What about https://docs.weblate.org/en/latest/admin/checks.html#customizing-behavior-using-flags especially https://docs.weblate.org/en/latest/user/checks.html#check-begin-space ?
Who of us has the privileges to change that?

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: I still do not think that this should be changed in weblate. You should not need to worry in English about line breaks either. I suggest to leave it as a memory and motivation to eventually fix this in the LinuxCNC code.

Comment (hansu) [reply src/po/gmoccapy/de.po:137]: Warum nicht? Ich meine weblate kann ja ruhig eine Waring anzeigen, wenn die Leerzeichen am Anfang nicht uebereinstimmen. Aber einfach den String wieder zurueckzuändern, nachdem man die Leerzeichen entfernt hat? Nee, das geht gar nicht! Hast du das selber mal probiert die Leerzeichen zu entfernen, damit zu versteht wovon ich rede?

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: Klingt wirklich komisch. Ich meine aber das ging. Hattest Du das "dismiss" weggeklickt und ein zweites mal gesichert?

Comment (hansu) [reply src/po/gmoccapy/de.po:137]: Ja, aber das bringt nichts. Immer wenn ich "Save and stay" klicke, sind die Leerzeichen wieder da.

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: Habe gerade diesen Eintrag bei weblate gesucht aber nicht gefunden. Bin dann alle Übersetzungen mit "offset" durchgegangen (54 sind es) und habe dieses dann je nach Kontext und Gefühl mit "Korrektur", oder "Versatz" oder "Verschiebung" übersetzt. Bei einen oder zwei Übersetzungen war ich etwas entsetzt - die waren völlig "off" (kleines Wortspiel), auch technisch, erwarteten etwa noch einen "%s" Parameter.  Da müssen wir uns echt noch ein paar Gedanken machen, schließlich kann es richtig teuer werden, wenn bei irgendeiner Aktualisierung von einer Einheit eine fuzzy Übersetzung bspw den Sprung von m zu mm nicht mitmacht.

Comment (hansu) [reply src/po/gmoccapy/de.po:137]: Bei (https://hosted.weblate.org/translate/linuxcnc/gmocappy/de/?checksum=ce5962f4d3b95c6)
```
"    Gewählten\n"
"Offset setzen"
```

hab ich das Problem ja dadurch umgangen, dass ich das Wort "Offset" hinzugefuegt habe. Somit ist die Zweile Zeile wieder länger als die erste. Wenn auch suboptimal da insgesamt etwas zu lang (breit).

Hier ist das andere Beispiel:
https://hosted.weblate.org/translate/linuxcnc/gmocappy/de/?checksum=618e73799f218873

Naja die Fuzzy-Uebersetzungen sind ja nur Vorschläge die entweder automatisch generiert werden oder manuell als fuzzy gesetzt werden. Da diese Strings aber nicht verwendet werden, sehe ich das nicht als problematisch an, was du beschrieben hast.

Wer ist denn unser Weblate-Admin? Macht Petter das?

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: Ne, vermutlich weiß silopolis da mehr.
Habe aus Deinem zweiten Beispiel gerade
```
    Nulle
Drehachsen
```
gemacht - nun hängt er da irgendwie bei weblate, vermutlich weil Du auf derselben Seite bist und gab mir einen internen Fehler. Crashed :)

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: Da ich bei den wenigstens Meldungen eine Idee habe, wie diese auf dem Bildschirm aussieht und wieviel Platz da jeweils in den Fenstern vorgesehen ist, muss es bei all unseren Mühen eine zweite Durchsicht geben, die sich dann der Optik annimmt. Ich möchte mir diese Leerzeichen-Gedanken derzeit nicht machen.

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: Eigentlich wollte ich mich primär um die Übersetzung der Handbücher mit kümmern, damit der Einstieg für alle leichter wird.

File: src/po/gmoccapy/de.po (hunk 1 above)
Comment (hansu) [reply src/po/gmoccapy/de.po:1921]: This is actually what "Rotation of Z" is doing. Here a value of 45 degrees:

grafik

File: src/po/gmoccapy/de.po (hunk 2 above)
Comment (hansu) [reply src/po/gmoccapy/de.po:137]: >Da ich bei den wenigstens Meldungen eine Idee habe, wie diese auf dem Bildschirm aussieht und wieviel Platz da jeweils in den Fenstern vorgesehen ist,

Deshalb beschränke ich mit bei den Uebersetzungen i.d.R. auf diese, wo ich den Kontext kenne...

Comment (hansu) [reply src/po/gmoccapy/de.po:137]: Will fit this case, but looks not very nice:

 grafik

Comment (hansu) [reply src/po/gmoccapy/de.po:137]: > Habe aus Deinem zweiten Beispiel gerade
> 
> ```
>     Nulle
> Drehachsen
> ```
Ja aber imperativ ist ja fuer sowas nicht so schön bzw. nicht gebräuchlich. Hattest du meinen Kommentar dazu in einer frueheren Uebersetzung gesehen?

Wie auch immer, Drehachse ist in diesem Kontext eh nicht korrekt. Man nullt damit die "Drehung um Z"

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: Danke für diese visuellen Eindrücke. Ich kann diese Arbeit nur nicht leisten. Möchte lieber einmal mit den Übersetzungen durch sein und dann die Programme und Fenster nacheinander kontrollieren. Derzeit weiß ich nicht einmal, ob ein Text auf der Konsole oder im Fenster oder als button erscheint. Aber deswegen können wir ja dennoch schonmal an Deinem Beispiel üben.

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: Wir können uns auch überlegen, verschiedene Screenshots automatisch erzeugen zu lassen. Das könnte für die Dokumentation sehr interessant sein.

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: "Drehung um Z" wäre m.E. dann richtig.

Comment (hansu) [reply src/po/gmoccapy/de.po:137]: > Ich kann diese Arbeit nur nicht leisten.

Musst du ja gar nicht. Es reicht sicher schon, wenn jeder die Oberfläche(n) uebersetzt, die er auch benutzt.

>Wir können uns auch überlegen, verschiedene Screenshots automatisch erzeugen zu lassen. Das könnte für die Dokumentation sehr interessant sein.

Klingt interessant. Stell ich mir aber nicht ganz trivial vor auf einem Buildserver ohne X-Server...

Comment (smoe) [reply src/po/gmoccapy/de.po:137]: > > "Drehung um Z" wäre m.E. dann richtig.
> 
> Genau. Hatte ich schon geändert.
> 
> > Ich kann diese Arbeit nur nicht leisten.
> 
> Musst du ja gar nicht. Es reicht sicher schon, wenn jeder die Oberfläche(n) uebersetzt, die er auch benutzt.

Sehe das anders. Die ganzen Fehlermeldungen bspw können wir ja nicht alle generieren. Wir müssen m.E. einmal durch, ich setze das "needs edit" recht häufig zu Unterstützung des peer review, und dann muss es tatsächlich wie von Dir vorgeführt überprüft werden.

> > Wir können uns auch überlegen, verschiedene Screenshots automatisch erzeugen zu lassen. Das könnte für die Dokumentation sehr interessant sein.
> 
> Klingt interessant. Stell ich mir aber nicht ganz trivial vor auf einem Buildserver ohne X-Server...

Dachte da an Xnest. Irgendwie klappt das, wenn wir es wirklich brauchen.

**Pull request merged**